addons/account/models/account_move.py

```diff
@@ -39,6 +39,16 @@ class AccountMove(models.Model):
     _check_company_auto = True
     _sequence_index = "journal_id"
 
+    def init(self):
+        self.env.cr.execute("""
+            CREATE INDEX IF NOT EXISTS account_move_to_check_idx
+            ON account_move(journal_id, to_check) WHERE to_check = true
+        """)
+        self.env.cr.execute("""
+            CREATE INDEX IF NOT EXISTS account_move_payment_idx
+            ON account_move(journal_id, state, payment_state, move_type, date)
+        """)
```

Comment:
- the 2 CREATE INDEX could be done in a single execute
- sorry, I forgot to mention that according to my tests the value of `to_check` does not need to be indexed if it is already in the `WHERE` clause (that makes very little disk space difference in practice but it makes more sense when you read the index definition)

```suggestion
        self.env.cr.execute("""
            CREATE INDEX IF NOT EXISTS account_move_to_check_idx
            ON account_move(journal_id) WHERE to_check = true;
            CREATE INDEX IF NOT EXISTS account_move_payment_idx
            ON account_move(journal_id, state, payment_state, move_type, date);
        """)
```